README.md

```diff
@@ -319,6 +319,12 @@ If there's a git mirror path and `mount-checkout` is enabled, the (mirror path)[
 
 Default: `true`
 
+### `chown` (optional, boolean)
+
+Whether to chown mounted files to the Buildkite agent user before exiting, to ensure that Buildkite can clean them up. Has no effect on Windows or if `mount-checkout` is false.
```

Comment:
I would also add a comment here about this using the `busybox:latest` image as root so that it doesn't come as a surprise that another container is being run. Also, this will run even if the step fails  (because it is in the pre-exit hook) and may take a long time if there are a lot of files in the checkout folder, which may cause issues as there are limits to how long the agent waits until the processes are killed by the agent - specially when the step is cancelled. Also as this only affects the checkout folder and not any other volumes mounted, I would add a comment about the `propagate-uid-gid` option.

Something like:

```suggestion
Whether to `chown` mounted files to the Buildkite agent user before exiting, to ensure that the agent can clean them up.

Note that:
* it will run a `busybox:latest` container image as a root user
* that container will run after the command even if it fails or the step is cancelled
* will only run for your checkout folder (not any other volumes or mounts)
* if your checkout folder is of considerable size, it may add to the execution time
* has no effect on Windows or if `mount-checkout` is false

You may also want to consider using the `propagate-uid-gid` option instead.
```

Reply:
Done.